frame/system/src/lib.rs

```diff
@@ -1013,15 +1013,16 @@ impl<T: Config> Module<T> {
 		}
 	}
 
-	/// Remove temporary "environment" entries in storage.
+	/// Remove temporary "environment" entries in storage, compute the storage root and return the
+	/// resulting header for this block.
 	pub fn finalize() -> T::Header {
 		ExecutionPhase::kill();
 		ExtrinsicCount::kill();
 		AllExtrinsicsLen::kill();
 
-		let number = <Number<T>>::take();
-		let parent_hash = <ParentHash<T>>::take();
-		let mut digest = <Digest<T>>::take();
+		let number = <Number<T>>::get();
```

Comment:
Not sure if it's worth adding a comment here on why these aren't cleared. We have a similar comment at the end of this function.